index/pretty.ml

```diff
@@ -1,95 +1,45 @@
+(** This file contains useful printer, that are however of dubious
+    maintainability. Their result is used to be parsed afteward, it is not
+    printed but consumed as the basis for type-search. Because of this it is
+    sensitive code. *)
```

Comment:
I do not really understand this module, and which it could be deleted.

Reply:
(s/which/wish/ ?) (I don't understand the comment either)
I think this module original purpose/name drifted quite a bit now that Odoc exposes more functionality to enable search:
- It was originally used to pretty-print types and value names, to display them on the website
- It looks like what remains (from its use by `load_doc`) is to convert a named type to its fullname, such that it can be indexed (eg the odoc identifier for `int` is converted to the string `"Stdlib.int"`, which is then ~parsed~ split on the character `'.'` to index both the short name `"int"` and the longer form `"Stdlib.int"` (to enable users to be more or less precise in their queries))

Perhaps Odoc exposes this "convert to fullname" functionality already? (in which case the module could be removed? or at least renamed?) (and perhaps the hacky "split on dot" could be done differently?)

The only tricky bit I remember is the handling of functors for the fullname expansion: Currently it produces `"Stdlib.Map.Make.t"` without parens to signal the `Make(...)` functor application. This was done to ease user queries (not having to type the parens), since we don't support restricting the functor argument (eg searching for a function which consumes a `Map.Make(Int).t`)

Reply:
I think the stdlib thing for core type is motivation enough to keep this, I will document it.